dashboard/visualise_time_changes.py

```diff
@@ -0,0 +1,80 @@
+"""Script to create visualisations of changes over time"""
+
+from datetime import datetime
+import streamlit as st
+import pandas as pd
+import altair as alt
+from db_functions import get_scores_topic, get_topic_names
+
+
+def resample_dataframe(df: pd.DataFrame, time_interval:str):
+    """Resamples the dataframe to return the average sentiment scores by source, topic 
+    over a set of grouped time intervals"""
+
+    df['date_published'] =pd.to_datetime(df['date_published'])
+
+    df_avg = df.groupby(['source_name', 'topic_name']).resample(time_interval, on= 'date_published').mean().reset_index()
+    return df_avg
+
+def generate_warning_message(source_to_topics: dict) -> str:
```

Comment:
Great idea to generating a warning message, will make the dashboard a lot easier to understand.